src/ReportManager/SchemaForm/SchemaFormContext/index.js

```diff
@@ -0,0 +1,52 @@
+import React, { createContext } from 'react';
+
+import { FORM_FIELDS_TYPES } from '../constants';
+import { textFieldDetailsFactory } from '../fields/fieldDetailsFactory';
+import { isFieldRequired } from '../utils';
+import {cloneDeep} from "lodash-es";
+
+export const SchemaFormContext = createContext(null);
+
+
+const SchemaFormContextProvider = ({ schema, onFormChange, formData, formErrors, children }) => {
+
+  const onFieldChange = (field, value) => {
+    onFormChange({
```

Comment:
This doesn't look like it'll work for collections/nested field data

Reply:
Same as [here](https://github.com/PADAS/das-web-react/pull/1186#discussion_r1844336035) I did not wanted to have a wider scope for this ticket, once the foundations and responsibilities of the pattern are set can be extended in the particular ticket